### ISSUE TITLE
Analysis not working for code imported with include macro

### ISSUE THREAD

**Issue by arturkow2000** [intellij-rust/intellij-rust#6165]
## Environment

* **IntelliJ Rust plugin version: 0.3.131.3366-202**
* **Rust toolchain version: nightly 2020-09-13**
* **IDE name and version: CLion 2020.2**
* **Operating system: Windows 10**

## Problem description
Code analysis is not working for code imported using include macro, rendering code completion unusable with any dynamically generated code.

## Steps to reproduce

```rust
mod generated {
   include!(concat!(env!("OUT_DIR"), "/generated.rs"));
}

// IDE won't detect anything in generated.rs
```

**Comment by Undin** [comment]
@arturkow2000 By default, the plugin doesn't know value of `OUT_DIR` env variable. It leads to unknown name for name resolution mechanism.
Try to enable `org.rust.cargo.evaluate.build.scripts` experimental feature (`Help | Find Action` and type `Experimental Features`). After that, each import of project data from Cargo (aka Cargo refresh) will launch `cargo check` to extract `OUT_DIR` for each crate. See #4734 for more details
Note, it may significantly increase time of Cargo refresh operation

**Comment by snaggen** [comment]
I have a similar issue, where I generate code that constructs enums and that code is then added to the project using an include! macro. And the enums are not listed for auto-completion, so my guess is that it is not indexed. 
May also be related to #6t164 that macro indexing is broken in general somehow.

**Comment by Undin** [comment]
@snaggen do you use `include` macro to attach generated code to your project? If yes, did you try `org.rust.cargo.evaluate.build.scripts` experimental feature described in https://github.com/intellij-rust/intellij-rust/issues/6165#issuecomment-699534975?

**Comment by snaggen** [comment]
Yes, I enabled that and tested, but no change. 
//Mattias Eriksson



8 okt. 2020 09:42 av notifications@github.com:

**Comment by Undin** [comment]
@snaggen could you share your code (or any other code example) where it doesn't work, please?

**Comment by snaggen** [comment]
Will try to create an example... but not until next week

**Comment by Undin** [comment]
@snaggen Great! Thank you for example. I'll take a look

**Comment by Undin** [comment]
@snaggen your issue is related neither to this issue nor to #6164.
I've creates a separate issue about your problem #6243

**Comment by Undin** [comment]
I think this issue can be closed
